Pr stage2unpacker 80x

mulhearn opened the pull request:
This PR adds all of the L1T raw data packers and unpackers that use the L1TRawToDigi framework.  It Includes the most crucial unpackers needed for the 2016 run (Calo Stage2, uGMT, and uGT).  It includes also the EMTF and BMTF unpacker, which are not yet included in the standard sequence.

This PR will become smaller as #13065 is merged.

matz-e committed:
Remove misleading warnings.

The code was changed a while back to accomodate cable splitting, unpacking
with fixed positions.  The warnings pertain to old code, where the unpacked
quantities were just pushed back onto the collection.

mulhearn committed:
Muon unpacker

mulhearn committed:
Stage-2 unpacker update

mulhearn committed:
l1t standard sequences

Conflicts:
	Configuration/StandardSequences/python/RawToDigi_cff.py
	L1Trigger/Configuration/python/SimL1Emulator_cff.py

mulhearn committed:
update rawtodigi digitoraw standard sequences

Conflicts:
	Configuration/StandardSequences/python/DigiToRaw_cff.py
	Configuration/StandardSequences/python/RawToDigi_cff.py

jimbrooke committed:
update packer block size and location

mulhearn committed:
bugfix from rename gimGmtDigis

Conflicts:
	L1Trigger/L1TCommon/test/runStandardSequences.py
	L1Trigger/L1TGlobal/python/simDigis_cff.py

mulhearn committed:
stage-2 unpacker update

mulhearn committed:
add uGT packer/unpacker to standard sequence

Conflicts:
	L1Trigger/Configuration/python/L1TDigiToRaw_cff.py
	L1Trigger/Configuration/python/L1TRawToDigi_cff.py

cmsbuild commented:
A new Pull Request was created by  @mulhearn  for CMSSW_8_0_X.

It involves the following packages:

CondCore/L1TPlugins
CondFormats/DataRecord
CondFormats/L1TObjects
DataFormats/L1TGlobal
DataFormats/L1TMuon
DataFormats/L1Trigger
EventFilter/L1TRawToDigi
EventFilter/RawDataCollector
L1Trigger/L1TCalorimeter
L1Trigger/L1TGlobal
L1Trigger/L1TMuon
L1Trigger/L1TMuonBarrel


The following packages do not have a category, yet:

DataFormats/L1TMuon
L1Trigger/L1TMuon
L1Trigger/L1TMuonBarrel

@diguida, @emeschi, @cerminar, @cmsbuild, @ggovi, @franzoni, @mommsen, @mmusich, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald, @apfeiffer1, @tocheng this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

mulhearn commented:
please test

mulhearn commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 52b6b06e25e612cea72151b7eafab119c7b4c386
 When I ran the RelVals I found an error in the following worklfows: 
 8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>
9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
101.0 step1
<pre>runTheMatrix-results/101.0_SingleElectronE120EHCAL+SingleElectronE120EHCAL/step1_SingleElectronE120EHCAL+SingleElectronE120EHCAL.log

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>
50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10736/summary.html

mulhearn committed:
bugfixes and updates post merge with dev-recipe

Conflicts:
	CondFormats/L1TObjects/interface/L1TMuonGlobalParams.h
	CondFormats/L1TObjects/src/L1TMuonGlobalParams.cc
	L1Trigger/L1TMuon/plugins/L1TMicroGMTLUTDumper.cc
	L1Trigger/L1TMuon/plugins/L1TMuonGlobalParamsESProducer.cc
	L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc
	L1Trigger/L1TMuon/src/MicroGMTCancelOutUnit.cc
	L1Trigger/L1TMuon/test/runMuon.py
	L1Trigger/L1TMuonBarrel/src/L1MuBMTrackFinder.cc

cmsbuild commented:
Pull request #13068 was updated. @diguida, @emeschi, @cerminar, @cmsbuild, @ggovi, @franzoni, @mommsen, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

mulhearn commented:
+1

cmsbuild commented:
-1 
Tested at: 52975cc944e01d90831534e50605dcba1c223281
 When I ran the RelVals I found an error in the following worklfows: 
 8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>
9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
101.0 step1
<pre>runTheMatrix-results/101.0_SingleElectronE120EHCAL+SingleElectronE120EHCAL/step1_SingleElectronE120EHCAL+SingleElectronE120EHCAL.log

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10799/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/d428e5941d534e6198f2e0456fde9c7f0be9b499
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10799/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10799/git-merge-result

abrinke1 committed:
Initial bits of MTF7 support from Matthias Wolf

gflouris committed:
Rpc bit in L1MuDTChambPhDigi

abrinke1 committed:
v1 of EMTF unpacking

Conflicts:
	DataFormats/L1TMuon/interface/EMTF/Counters.h
	DataFormats/L1TMuon/interface/EMTF/EventHeader.h
	DataFormats/L1TMuon/interface/EMTF/SP.h
	DataFormats/L1TMuon/interface/EMTFOutput.h

eparadas committed:
all BMTF unpacker files are added and tested

eparadas committed:
removing cout statements

eparadas committed:
adding all the input links

abrinke1 committed:
v1 of EMTF unpacking, continued

Conflicts:
	DataFormats/L1TMuon/interface/EMTF/ME.h

cmsbuild commented:
Pull request #13068 was updated. @diguida, @cerminar, @cmsbuild, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 04441722ddf9e337963ce6865e8fba939d3f2f1d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10914/summary.html

mulhearn commented:
+1

cmsbuild commented:
+1
Tested at: 04441722ddf9e337963ce6865e8fba939d3f2f1d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10920/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10920/summary.html

mulhearn commented:
I can rebase to remove ALCA/DB now...

mulhearn committed:
bugfix for Stage-1 unpacking

Conflicts:
	L1Trigger/L1TCommon/plugins/L1TSummary.cc

mulhearn committed:
cleanup unused header, commented code blocks

cmsbuild commented:
Pull request #13068 was updated. @cmsbuild, @mulhearn, @davidlange6 can you please check and sign again.

mulhearn commented:
(Rebased, no more ALCA/DB sign off needed)

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 173eacd4c718d4dbb3582664a9d6a616c54438ac
 When I ran the RelVals I found an error in the following worklfows: 
 134.911 step2
<pre>runTheMatrix-results/134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step2_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT.log

----- Begin Fatal Exception 04-Feb-2016 10:07:16 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Validating configuration of module: class=L1TRawToDigi label='hltCaloStage1Digis'

Exception Message:

In the configuration, parameter "CTP7" is defined as an untracked bool.

It should be tracked.

----- End Fatal Exception -------------------------------------------------

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

----- Begin Fatal Exception 04-Feb-2016 10:15:33 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Validating configuration of module: class=L1TRawToDigi label='hltCaloStage1Digis'

Exception Message:

In the configuration, parameter "CTP7" is defined as an untracked bool.

It should be tracked.

----- End Fatal Exception -------------------------------------------------

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

----- Begin Fatal Exception 04-Feb-2016 10:19:12 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Validating configuration of module: class=L1TRawToDigi label='hltCaloStage1Digis'

Exception Message:

In the configuration, parameter "CTP7" is defined as an untracked bool.

It should be tracked.

----- End Fatal Exception -------------------------------------------------

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

----- Begin Fatal Exception 04-Feb-2016 10:27:05 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Validating configuration of module: class=L1TRawToDigi label='hltCaloStage1Digis'

Exception Message:

In the configuration, parameter "CTP7" is defined as an untracked bool.

It should be tracked.

----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10989/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/0a7650d9c75a3b1fc41290f9ec03db56e2c57e14
https://github.com/cms-sw/cmssw/commit/bcb909dfd9c35999f6a1cbfc08cb10b6e26f0f47
https://github.com/cms-sw/cmssw/commit/c69dfd047d9ccf0e5b48278ac9fc0287dbc8cf0a
https://github.com/cms-sw/cmssw/commit/bf5c902359e10755886aea190bdfe13938211105
https://github.com/cms-sw/cmssw/commit/65fd61fc4bc439e90a8fab1d77e7b3325a5c5877
https://github.com/cms-sw/cmssw/commit/bc5d94b6d374e3740079d7facc9580b696a7f777
https://github.com/cms-sw/cmssw/commit/195b670807efe11d491bf903a37f602ac71f2a1e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10989/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/10989/git-merge-result

cmsbuild commented:
Pull request #13068 was updated. @cmsbuild, @mulhearn, @davidlange6 can you please check and sign again.

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

mulhearn committed:
remove data and manual conflict resolution

mulhearn committed:
manual sync to dev-recipes

mulhearn committed:
switch some params to untracked to avoid breaking HLT

cmsbuild commented:
Pull request #13068 was updated. @mommsen, @cmsbuild, @emeschi, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: 12d7411eefb6be1d06c5f913c9bf26402e2afb4b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11002/summary.html

mulhearn committed:
Merged pr-stage2unpacker-80x from repository cms-l1t-offline

mulhearn committed:
restore EventFilter/RawDataCollector/python/rawDataCollector_cfi.py

cmsbuild commented:
Pull request #13068 was updated. @cmsbuild, @mulhearn, @davidlange6 can you please check and sign again.

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11002/summary.html

cmsbuild commented:
+1
Tested at: 6e5b2b54b519654c0c800ea995f13c36112b9b40
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11009/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/ab93d0ca2b017b0e85a5f91edeb76824dbbc1a46
https://github.com/cms-sw/cmssw/commit/08a989ffaa96b7582a5ced05bc3dca8aa0a8ac36
https://github.com/cms-sw/cmssw/commit/6129747c5afb12be429d65cdd6db83f1ff33dcbc
https://github.com/cms-sw/cmssw/commit/7c335d88f6790902da71ad2ff37a75690d5b86c9
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11009/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11009/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11009/summary.html

mulhearn commented:
+1

mulhearn commented:
@davidlange6  FYI this one is the next in the L1 pipeline.  Hahhaha... thats a DAQ joke.  Get it?

davidlange6 commented on DataFormats/L1DTTrackFinder/interface/L1MuDTChambPhDigi.h:
```diff
@@ -41,6 +41,9 @@ class L1MuDTChambPhDigi {
 
+  L1MuDTChambPhDigi( int ubx, int uwh, int usc, int ust,
+        int uphr, int uphb, int uqua, int utag, int ucnt, int urpc);
+
```
@mulhearn - remove the old constructor and add a default of -10 for int urpc

mulhearn replied:
OK, will update.

davidlange6 commented on EventFilter/L1TRawToDigi/python/caloStage1Raw_cfi.py:
```diff
@@ -3,13 +3,13 @@
+    HFBitCountsInputLabel = cms.InputTag("simCaloStage1FinalDigis", "HFBitCounts"),
+    HFRingSumsInputLabel = cms.InputTag("simCaloStage1FinalDigis", "HFRingSums"),
+    RegionInputLabel = cms.InputTag("simRctDigis", ""),
+    EmCandInputLabel = cms.InputTag("simRctDigis", ""),
```
@mulhearn - why this change?

mulhearn replied:
Actually, that looks wrong... why didn't this break anything?

mulhearn replied:
Our open development branch model leads to some surprises like this slipping by...

mulhearn replied:
Sorry, no actually, that is a correct correction.  We produce RAW from simulated digis (or the detector does it for us).  So the old version was a mistake.  This one is correct.

davidlange6 commented on EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py:
```diff
@@ -5,7 +5,7 @@
-    FWId            = cms.untracked.int32(2),
+    FWId            = cms.uint32(2),
     lenSlinkHeader  = cms.untracked.int32(8),
     lenSlinkTrailer = cms.untracked.int32(8),
```
likely all of these parameters should be tracked..

mulhearn replied:
Indeed... but my attempt to changing some of the other ones broke HLT scripts.  So was postponing this until HLT/L1T integration step.

mulhearn commented:
Let's hold off on this one until I can make your suggested changes...  give me a few.

mulhearn committed:
cleanup ctor for L1MuDTChambPhDigi

mulhearn commented:
OK, there's the update.

mulhearn commented:
+1

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #13068 was updated. @cmsbuild, @davidlange6 can you please check and sign again.

cmsbuild commented:
-1 
Tested at: 6be1f23ca55753817200b74283bcae589ac7c3f3
 I found an error when building: 
 
 <pre>>> Compiling python modules src/L1Trigger/L1TMuonEndCap/python
>> Compiling python modules src/DQMServices/Components/python
>> All python modules compiled
>> Pluging of all type refreshed.
gmake[1]: Leaving directory `/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-04-2300'
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11025/summary.html

mulhearn commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 6be1f23ca55753817200b74283bcae589ac7c3f3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11031/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13068/11031/summary.html
